### PR TITLE
Fix EGL OpenGLControlBase on Win32

### LINKED ISSUE #12051: InvalidOperationException:Unable to locate IPlatformGraphicsOpenGlContextFactory
**Describe the bug**
I just make the control based on OpenGL
it worked in 11.0.0-RC1.1.
but it stopped working in 11.0.0-RC2.2 and 11.0.0

**To Reproduce**

```
[OpenGL]Unable to initialize OpenGL: 'System.InvalidOperationException: Unable to locate 'Avalonia.OpenGL.IPlatformGraphicsOpenGlContextFactory'.
   at Avalonia.LocatorExtensions.GetRequiredService[T](IAvaloniaDependencyResolver resolver) in /_/src/Avalonia.Base/AvaloniaLocator.cs:line 140
```

**Expected behavior**
Render without any problems

**Desktop:**

- OS: Windows
- Version 11.0.0

### PR DESCRIPTION
**Pull request by maxkatz6**
## Fixed issues

Fixes #12051

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0037413-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

File: src/Windows/Avalonia.Win32/OpenGl/WglDisplay.cs [hunk 1]
```diff
@@ -144,8 +142,12 @@ private static void DebugCallback(int source, int type, int id, int severity, in
                             });
                     }
 
-                    using(new WglRestoreContext(dc, context, null))
-                        s_glDebugMessageCallback(Marshal.GetFunctionPointerForDelegate(_debugCallback), IntPtr.Zero);
+                    if (s_glDebugMessageCallback is not null)
+                    {
+                        using (new WglRestoreContext(dc, context, null))
```
Comment (jp2masa) [inline src/Windows/Avalonia.Win32/OpenGl/WglDisplay.cs:147]: This `using` is no longer affecting the `return`, is that intentional?

Comment (maxkatz6) [reply src/Windows/Avalonia.Win32/OpenGl/WglDisplay.cs:147]: It wasn't affecting it before, was it?
I.e. 
```
                    using(new WglRestoreContext(dc, context, null))
                        s_glDebugMessageCallback(Marshal.GetFunctionPointerForDelegate(_debugCallback), IntPtr.Zero);
```
using here was disposed right after s_glDebugMessageCallback exected.
It's still the same, but this context is captured only when s_glDebugMessageCallback is not null

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0037440-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

File: src/Windows/Avalonia.Win32/OpenGl/WglDisplay.cs (hunk 1 above)
Comment (jp2masa) [reply src/Windows/Avalonia.Win32/OpenGl/WglDisplay.cs:147]: Sorry, I read it wrong.

**Review by kekekeks — APPROVED** [review]

**Pull request merged**